subsys/bluetooth/controller/hci/hci.c

```diff
@@ -4974,6 +4986,168 @@ struct net_buf *hci_vs_err_assert(const char *file, uint32_t line)
 }
 #endif /* CONFIG_BT_HCI_VS_FATAL_ERROR */
 
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
+static void vs_le_df_connectionless_iq_report(struct pdu_data *pdu_rx, struct node_rx_pdu *node_rx,
+					      struct net_buf *buf)
+{
+	struct bt_hci_evt_vs_le_connectionless_iq_report *sep;
+	struct node_rx_iq_report *iq_report;
+	struct lll_sync *lll;
+	uint8_t samples_cnt;
+	int16_t rssi;
+	uint16_t sync_handle;
+	uint16_t per_evt_counter;
+	struct ll_sync_set *sync = NULL;
+
+	iq_report = (struct node_rx_iq_report *)node_rx;
+
+	if (!(vs_events_mask & BT_EVT_MASK_VS_LE_CONNECTIONLESS_IQ_REPORT)) {
+		return;
+	}
+
+	lll = iq_report->hdr.rx_ftr.param;
+
+	sync = HDR_LLL2ULL(lll);
+
+	/* TX LL thread has higher priority than RX thread. It may happen that
+	 * host successfully disables CTE sampling in the meantime.
+	 * It should be verified here, to avoid reporting IQ samples after
+	 * the functionality was disabled or if sync was lost.
+	 */
+	if (ull_df_sync_cfg_is_not_enabled(&lll->df_cfg) || !sync->timeout_reload) {
+		/* Drop further processing of the event. */
+		return;
+	}
+
+	/* Get the sync handle corresponding to the LLL context passed in the
+	 * node rx footer field.
+	 */
+	sync_handle = ull_sync_handle_get(sync);
+	per_evt_counter = iq_report->event_counter;
+
+	/* If there are no IQ samples due to insufficient resources
+	 * HCI event should inform about it by storing single octet with
+	 * special I_sample and Q_sample data.
+	 */
+	samples_cnt = (!iq_report->sample_count ? 1 : iq_report->sample_count);
```

Comment:
Below you are doing 
```c
samples_cnt = MAX(1, iq_report->sample_count);
```
instead.

Either way of course works, but might make sense to do it the same way both places :)